env0/utils.go

```diff
@@ -221,6 +221,94 @@ func writeResourceData(i interface{}, d *schema.ResourceData) error {
 	return nil
 }
 
+func getInterfaceValues(i interface{}) []interface{} {
+	var result []interface{}
+
+	val := reflect.ValueOf(i)
+
+	for i := 0; i < val.Len(); i++ {
+		field := val.Index(i)
+		result = append(result, field.Interface())
+	}
+
+	return result
+}
+
+// Extracts values from a slice of interfaces, and writes it to resourcedata at name.
+func writeResourceDataSlice(i interface{}, name string, d *schema.ResourceData) error {
+	writeResourceDataSliceStruct := func(val reflect.Value, value map[string]interface{}) {
```

Comment:
This add supports structs within structs. Hopefully, this is enough to cover the vast majority of use cases.